### PR TITLE
txnprovider/txpool: move more txpool types out of erigon-lib

### PR DESCRIPTION
**Pull request by taratorio**
move last bits of txpool related types out of erigon-lib
in subsequent PRs will further split and organise some of these into better packages

### CODE REVIEW

File: erigon-lib/state/aggregator_files.go [hunk 1]
```diff
@@ -150,106 +144,3 @@ func (mf MergedFiles) Close() {
 		}
 	}
 }
-
```
Comment (taratorio) [inline erigon-lib/state/aggregator_files.go:153]: unused code

File: erigon-lib/state/aggregator_files.go [hunk 2]
```diff
@@ -150,106 +144,3 @@ func (mf MergedFiles) Close() {
...
-func DecodeAccountBytes(enc []byte) (nonce uint64, balance *uint256.Int, hash []byte) {
```
Comment (taratorio) [inline erigon-lib/state/aggregator_files.go:154]: unused code

File: erigon-lib/state/aggregator_files.go [hunk 3]
```diff
@@ -150,106 +144,3 @@ func (mf MergedFiles) Close() {
...
-		pos++
-		if codeHashBytes > 0 {
-			codeHash := make([]byte, length.Hash)
-			copy(codeHash, enc[pos:pos+codeHashBytes])
-		}
-	}
-	return
-}
-
-func EncodeAccountBytes(nonce uint64, balance *uint256.Int, hash []byte, incarnation uint64) []byte {
```
Comment (taratorio) [inline erigon-lib/state/aggregator_files.go:181]: unused code

File: erigon-lib/types/testdata.go [hunk 4]
```diff
@@ -1,98 +0,0 @@
...
-//
-// You should have received a copy of the GNU Lesser General Public License
-// along with Erigon. If not, see <http://www.gnu.org/licenses/>.
-
-package types
-
-import (
-	"github.com/holiman/uint256"
-)
-
```
Comment (taratorio) [inline erigon-lib/types/testdata.go:22]: moved all of this to txnprovider/txpool/test_data_test.go

**Review by AskAlexSharov — APPROVED** [review]

File: core/types/transaction_test.go [hunk 5]
```diff
@@ -738,89 +734,21 @@ func TestBlobTxEncodeDecode(t *testing.T) {
...
-	err = wrappedBlobTx.DecodeRLP(rlp.NewStream(bytes.NewReader(blobTxRlp[1:]), 0))
-	if err != nil {
-		t.Errorf("long rlp decoding failed: %v", err)
-	}
-	assertEqual(bbtx, &wrappedBlobTx.Tx)
-}
-
-// This test actually applies to testing the behaviour as seen from the
-// onNewTx behaviour in filters.go
-func TestShortUnwrapLib(t *testing.T) {
```
Comment (taratorio) [inline core/types/transaction_test.go:810]: this test and `TestShortUnwrap` are 1:1 the same now

File: core/types/transaction_test.go [hunk 6]
```diff
@@ -738,89 +734,21 @@ func TestBlobTxEncodeDecode(t *testing.T) {
 	}
 }
 
-func makeBlobTxRlp() []byte {
```
Comment (taratorio) [inline core/types/transaction_test.go:741]: moved to core/types/typestest pkg

**Pull request merged**